io/io/src/TFileBufferRead.cxx

```diff
@@ -0,0 +1,165 @@
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <string>
+
+#ifndef WIN32
+#include <sys/types.h>
+#include <sys/mman.h>
+#endif
+
+#include "TFile.h"
+#include "TError.h"
+#include "TSystem.h"
+
+#include "TFileBufferRead.h"
+
+
+// Download files in 128MB chunks.
+#define CHUNK_SIZE (128*1024*1024)
+
+
+TFileBufferRead::TFileBufferRead (TFile *file) :
+   fFile(file)
+{
+   fSize = fFile->GetSize();
+   fTotal = (fSize + CHUNK_SIZE - 1) / CHUNK_SIZE;
+   fPresent.resize(fTotal, 0);
+
+   if (TmpFile()) {
+      fInvalid = false;
+   }
+#ifdef WIN32
+   fInvalid = true;
+#endif
+}
+
+
+TFileBufferRead::~TFileBufferRead() {
+   if (fFd >= 0) {close(fFd);}
+}
+
+///
+// Create a temporary file on disk and unlink it; use it as a buffer
+// for the source TFile.
+//
+// - tmpdir: Directory where the temporary file should be created.
+//
+// Returns false on failure.
+//
+Bool_t TFileBufferRead::TmpFile(const std::string &tmpdir) {
+   TString base = "root-shadow-";
+   FILE *fp = gSystem->TempFileName(base, tmpdir.empty() ? nullptr : tmpdir.c_str());
+   if (!fp) {
+      Warning("TFileBufferRead", "Cannot create temporary file %s: %s (errno=%d)",
+                                 base.Data(), strerror(errno), errno);
+      return false;
+   }
+   int fd = fileno(fp);
+   if (-1 == unlink(base.Data())) {
+      Warning("TFileBufferRead", "Cannot unlink temporary file %s: %s (errno=%d)",
+                                 base.Data(), strerror(errno), errno);
+      return false;
+   }
+   if (-1 == ftruncate(fd, fSize)) {
+      Warning("TFileBufferRead", "Cannot resize temporary file %s: %s (errno=%d)",
+                                 base.Data(), strerror(errno), errno);
+      // We ignore this error - it shouldn't be fatal.
+   }
+   fFd = fd;
+   return true;
+}
+
+
+///
+// Positional read of the file, using the underlying buffer file.
+//
+// If the buffer file is invalid, then this reads from the underlying file.
+//
+// On error, -1 is returned and errno is set appropriately.
+Long_t TFileBufferRead::Pread(char *into, UInt_t n, Long64_t pos) {
```

Comment:
Isnt't 'PRead' or more accurate casing?

Reply:
Actually is the 'p' really useful.  I.e. is there any type of 'Read' for a TFileBufferRead?